internal/resources/policy/type/security/scope_schema.go

```diff
@@ -0,0 +1,154 @@
+
+	if v, ok := scopeData[clusterKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopedFullnameData = &scopedFullname{
+				scope:           clusterScope,
+				fullnameCluster: scoperesource.ConstructClusterPolicyFullname(v1, name),
+			}
+		}
+	}
+
+	if v, ok := scopeData[clusterGroupKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopedFullnameData = &scopedFullname{
+				scope:                clusterGroupScope,
+				fullnameClusterGroup: scoperesource.ConstructClusterGroupPolicyFullname(v1, name),
+			}
+		}
+	}
+
+	if v, ok := scopeData[organizationKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopedFullnameData = &scopedFullname{
+				scope:                organizationScope,
+				fullnameOrganization: scoperesource.ConstructOrganizationPolicyFullname(v1, name),
+			}
+		}
+	}
+
+	return scopedFullnameData
+}
+
+func flattenScope(scopedFullname *scopedFullname) (data []interface{}, name string) {
+	if scopedFullname == nil {
+		return data, name
+	}
+
+	flattenScopeData := make(map[string]interface{})
+
+	switch scopedFullname.scope {
+	case clusterScope:
+		name = scopedFullname.fullnameCluster.Name
+		flattenScopeData[clusterKey] = scoperesource.FlattenClusterPolicyFullname(scopedFullname.fullnameCluster)
+	case clusterGroupScope:
+		name = scopedFullname.fullnameClusterGroup.Name
+		flattenScopeData[clusterGroupKey] = scoperesource.FlattenClusterGroupPolicyFullname(scopedFullname.fullnameClusterGroup)
+	case organizationScope:
+		name = scopedFullname.fullnameOrganization.Name
+		flattenScopeData[organizationKey] = scoperesource.FlattenOrganizationPolicyFullname(scopedFullname.fullnameOrganization)
+	}
+
+	return []interface{}{flattenScopeData}, name
+}
+
+func validateScope(ctx context.Context, diff *schema.ResourceDiff, i interface{}) error {
+	value, ok := diff.GetOk(scopeKey)
+	if !ok {
+		return fmt.Errorf("scope: %v is not valid: minimum one valid scope block is required", value)
+	}
+
+	data, _ := value.([]interface{})
+
+	if len(data) == 0 || data[0] == nil {
+		return fmt.Errorf("scope data: %v is not valid: minimum one valid scope block is required", data)
+	}
+
+	scopeData := data[0].(map[string]interface{})
+	scopesFound := make([]string, 0)
+	scopesAllowed := []string{clusterKey, clusterGroupKey, organizationKey}
+
+	if v, ok := scopeData[clusterKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopesFound = append(scopesFound, clusterKey)
+		}
+	}
+
+	if v, ok := scopeData[clusterGroupKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopesFound = append(scopesFound, clusterGroupKey)
+		}
+	}
+
+	if v, ok := scopeData[organizationKey]; ok {
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			scopesFound = append(scopesFound, organizationKey)
+		}
+	}
+
+	if len(scopesFound) == 0 {
+		return fmt.Errorf("no valid scope type block found: minimum one valid scope type block is required among: %v", scopesAllowed)
```

Comment:
can you added the delimiter between the elements of scopesAllowed

Reply:
and in the scopesFound

Reply:
Sure

Reply:
Updated @shreyassreenivas